crates/subspace-node/src/commands/run/consensus.rs

```diff
@@ -468,6 +469,7 @@ pub(super) fn create_consensus_chain_configuration(
             force_authoring = true;
             network_options.allow_private_ips = true;
             timekeeper_options.timekeeper = true;
+            override_sync_mode = true;
```

Comment:
You could have just done this instead:
```suggestion
            sync = ChainSyncMode::Full;
```

However, the issue is only happening in dev mode on genesis, Snap sync can succeed if there were some segments produced, so I think `sync` type needs to change to `Option<ChainSyncMode>` and then we can do this in to ensure not overriding whatever user has specified explicitly:
```suggestion
            if sync.is_none() {
                sync.replace(ChainSyncMode::Full);
            }
```

Reply:
>  Snap sync can succeed if there were some segments produced...

Is there a way to detect the chain state at this stage? I'd guess we will have either access to local storage or best hash from remote peers much later.

Reply:
Or did you mean that we change the type of the CLI argument `sync` to optional and we'll set the default override in dev mode when we need snap-sync?